.travis.yml

```diff
@@ -1,25 +1,56 @@
 sudo: required
-dist: trusty
-language: generic
-
-cache:
-  directories:
-    - $HOME/.ccache
 
 git:
   quiet: true
 
 env:
   global:
-    - CCACHE_DIR=$HOME/.ccache
     - ROS_REPO=ros
     - CMAKE_ARGS="-DENABLE_TESTS=ON"
-  matrix:
-    - ROS_DISTRO="kinetic"
-    - ROS_DISTRO="melodic"
+
+jobs:
+  include:
+    - os: linux
+      dist: trusty
+      language: cpp
+      env:
+      - ROS_DISTRO="kinetic"
+      - CCACHE_DIR=$HOME/.ccache
+      cache:
+        directories:
+          - $HOME/.ccache
+    - os: linux
+      dist: xenial
+      language: cpp
+      env:
+      - ROS_DISTRO="melodic"
+      - CCACHE_DIR=$HOME/.ccache
+      cache:
+        directories:
+          - $HOME/.ccache
+    - os: windows
+      language: cpp
 
 install:
-  - git clone --quiet --depth=1 https://github.com/ros-industrial/industrial_ci.git .ci_config
+  - if [ "$TRAVIS_OS_NAME" != "windows" ]; then git clone --quiet --depth=1 https://github.com/ros-industrial/industrial_ci.git .ci_config; fi
+  - |
+    if [ "$TRAVIS_OS_NAME" = "windows" ]; then
+        unset CXX
+        unset CXX_FOR_BUILD
+        unset CC
+        unset CC_FOR_BUILD
+        choco feature disable -n=showDownloadProgress
+        cd ..
+        mkdir -p catkin_ws/src
+        mv opw_kinematics catkin_ws/src/
+        cd catkin_ws
+        choco uninstall mingw
+        choco source add -n=ros-win -s="https://roswin.azurewebsites.net/api/v2" --priority=1
+        choco install ros-melodic-ros_base -y --execution-timeout=0
```

Comment:
If we use `rosdep` later (in `travis_ci_windows.bat`), do we need to install `ros-melodic-ros_base` here?

Reply:
I will test it.

Reply:
I doubt installing `ros-melodic-ros_base` can be skipped. It installs a mini Linux runtime environment with a huge number of dependencies that outside of "ROS" itself.

Reply:
I tested it and it does not work because it could not find rosdep.

Reply:
You should be able to install  `rosdep` separately.

Reply:
The `ros_base` package installs its own version of Python separate from the system version along with a lot of other support libraries and utilities. Untangling that dependency chain is probably not worth the effort.